addons/mail/static/tests/core/record.test.js

```diff
@@ -1053,3 +1054,34 @@ test("insert with id relation keeps existing field values", async () => {
     expect(member2.channel.eq(channel1)).toBe(true);
     expect(member2.is_internal).toBe(true);
 });
+
+test("insert() different PY model having same JS model", async () => {
+    (class Thread extends Record {
+        static id = "id";
+    }).register(localRegistry);
+    patchWithCleanup(pyToJsModels, {
+        "discuss.channel": "Thread",
+        "mail.thread": "Thread",
+    });
+    const store = (await start2()).services["mail.store"];
```

Comment:
Why not just using start?

Reply:
Start returns the store itself, not the service.

Reply:
The `store.insert()` feature is in store_service, but I'd argue this should be moved to the internal and actual `store`.

It was still good to be cautious and have it in `store_service` while we were unsure of whether this should become standard store.

Not to do in this PR of course, just saying what I think about `store.insert()`

Reply:
I don't see what is so specific in this test that it needs another return value from start than the others.

Also the result of store service is the store record so it's exactly the same no?

Reply:
Ok now I understand, but then it should be a test in `store_service.test.js` like other similar tests, not in `record.test.js`.